src/search1_conversion/convert_params.py

```diff
@@ -202,16 +203,37 @@ def _get_search_params(params):
         sort.append({prop: {'order': order}})
     pagination = params.get('pagination', {})
     access_filter = params.get('access_filter', {})
-    with_private = bool(access_filter.get('with_private'))
-    with_public = bool(access_filter.get('with_public'))
+    with_private = access_filter.get('with_private')
+    with_public = access_filter.get('with_public')
+
+    if with_private is None and with_public is None:
+        only_public = False
+        only_private = False
+    else:
+        with_private = bool(with_private)
+        with_public = bool(with_public)
+        if with_private:
```

Comment:
I think this is all fine but I wonder if we flattened the different cases that would make it a little easier to follow.